cmake/ConfigureCompiler.cmake

```
@@ -89,16 +89,33 @@ else()
     set(GCC YES)
   endif()
 
+  # Use the linker environmental variable, if specified and valid
+  if ((USE_LD STREQUAL "DEFAULT") AND (NOT "$ENV{USE_LD}" STREQUAL ""))
+    string(TOUPPER "$ENV{USE_LD}" USE_LDENV)
+    if (("${USE_LDENV}" STREQUAL "LD") OR ("${USE_LDENV}" STREQUAL "GOLD") OR ("${USE_LDENV}" STREQUAL "LLD") OR ("${USE_LDENV}" STREQUAL "BFD"))
+      set(USE_LD "${USE_LDENV}")
+    endif()
```

```suggestion
    else()
      message (FATAL_ERROR "USE_LD must be set to DEFAULT, LD, GOLD, or LLD!")
    endif()
```

I would like to be specific. Otherwise, a user setting the env-variable to something invalid will not see an error but just unexpected default behavior.